flang/lib/Lower/IO.cpp

```diff
@@ -672,7 +672,8 @@ static void genIoLoop(Fortran::lower::AbstractConverter &converter,
   const auto &itemList = std::get<0>(ioImpliedDo.t);
   const auto &control = std::get<1>(ioImpliedDo.t);
   const auto &loopSym = *control.name.thing.thing.symbol;
-  mlir::Value loopVar = converter.getSymbolAddress(loopSym);
```

Comment:
getSymbolAddress() doesn't appear to be used all that often, but seems like it will not work in any situation where the symbol is a boxed variable.

Reply:
It is used in OpenMP/OpenACC lowering in a few other places. Filed an issue https://github.com/flang-compiler/f18-llvm-project/issues/1595 to capture this.